### PR TITLE
prevent long tap marker on OSM if map item got hit (fix #13168)

### LINKED ISSUE #13168: [UI] LongClick on cache behavior broken
### Describe your problem!

When performing a long click on a cache while using mapsforge map, a pin gets shown. However this pin should only appear when long-clicking on an empty location.

Screenshot_20220619_025235

### How to reproduce?

- Use Mapsforge
- long-click on a cache

=> ...an unexpected pin will appear. 

### Actual result after these steps?

Same as it works for our Google map. No pin should appear. 

### Expected result after these steps?

_No response_

### Reproducible

Yes

### c:geo Version

2022.06.19-NB

### System information

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by moving-bits**
## Description
Prevent displaying a map marker on long tap if an existing map item got hit (OSM only).

### CODE REVIEW

File: main/src/cgeo/geocaching/maps/mapsforge/v6/layers/TapHandlerLayer.java [hunk 1]
```diff
@@ -50,7 +50,9 @@ public boolean onTap(final LatLong tapLatLong, final Point layerXY, final Point
     public boolean onLongPress(final LatLong tapLatLong, final Point layerXY, final Point tapXY) {
         longTabLatLong = tapLatLong;
         requestRedraw();
-        tapHandler.onLongPress(tapXY);
+        if (!tapHandler.onLongPress(tapXY)) {
+            longTabLatLong = null;
+        }
```
Comment (fm-sys) [inline main/src/cgeo/geocaching/maps/mapsforge/v6/layers/TapHandlerLayer.java:55]: Works fine in the emulator. Although I think there should be a `requestRedraw()` after setting `longTabLatLong` to null. Or, why not inverting the if condition and setting the `longTapLatLong` only after the `onLongPress` had returned true?

File: main/src/cgeo/geocaching/maps/mapsforge/v6/TapHandler.java [hunk 2]
```diff
@@ -42,13 +42,16 @@ public synchronized void finished() {
         hitItems.clear();
     }
 
-    public void onLongPress(final Point tapXY) {
+    /** returns true if map item was hit */
```
Comment (fm-sys) [inline main/src/cgeo/geocaching/maps/mapsforge/v6/TapHandler.java:45]: > returns true if map item was hit

Isn't it exactly the other way around? If no item was hit / if empty space was hit...

Comment (moving-bits) [reply main/src/cgeo/geocaching/maps/mapsforge/v6/TapHandler.java:45]: you're right, will change the comment

File: main/src/cgeo/geocaching/maps/mapsforge/v6/layers/TapHandlerLayer.java (hunk 1 above)
Comment (moving-bits) [reply main/src/cgeo/geocaching/maps/mapsforge/v6/layers/TapHandlerLayer.java:55]: I need to set `longTapLatLong` before calling `tapHandler.onLongPress()` as it depends on a non-null value for it, but moving the `requestRedraw()` after the `if` clause seems to be ok. Updated PR will follow.

**Pull request merged**